src/components/button/index.js

```diff
@@ -1,29 +1,45 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 import { translate } from 'react-i18next';
-import style from './style.styl';
-
 import Modal from '../modal'
+import style from './style.styl';
 
 class Button extends Component {
+
+    static defaultProps = {
+        type: 'normal',
+        loading: false,
+        apply: () => {}
+    }
+
+    static propTypes = {
+        type: PropTypes.string.isRequired,
```

Comment:
i thinks `isRequired` means you **need** to pass it everytime, here we have it in default props